core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
I'm not convinced that keeping it here has any impact on whether or not when someone modifies `Environment` they are more or less likely to adapt this method. If that's a concern, we need something stronger than pushing it into `Environment` and increasing the public API. I'm still of the mind that this should be on `Node` and private.